RELEASE.md

```diff
@@ -7,6 +7,18 @@
 ## Breaking changes to the API
 Merged the ExistsMixin into AbstractDataSet
 
+# [Upcoming] Release 0.14.2
```

Comment:
The next release is 0.14.3. If you update the branch, you should get the latest `RELEASE.md`

Reply:
And Jo should be added as a contributor for the next release.